src/Model/Contact.php

```diff
@@ -94,6 +94,210 @@ class Contact extends BaseObject
 	 * @}
 	 */
 
+	/**
+	 * @brief Returns the contact id for the user and the public contact id for a given contact id
+	 *
+	 * @param int $cid Either public contact id or user's contact id
+	 * @param int $uid User ID
+	 *
+	 * @return array with public and user's contact id
+	 */
+	private static function getPublicAndUserContacID($cid, $uid)
+	{
+		if (empty($uid) || empty($cid)) {
+			return [];
+		}
+
+		$contact = DBA::selectFirst('contact', ['id', 'uid', 'url'], ['id' => $cid]);
+		if (!DBA::isResult($contact)) {
+			return [];
+		}
+
+		// We quit when the user id don't match the user id of the provided contact
+		if (($contact['uid'] != $uid) && ($contact['uid'] != 0)) {
+			return [];
+		}
+
+		if ($contact['uid'] != 0) {
+			$pcid = Contact::getIdForURL($contact['url'], 0, true, ['url' => $contact['url']]);
+			if (empty($pcid)) {
+				return [];
+			}
+			$ucid = $contact['id'];
+		} else {
+			$pcid = $contact['id'];
+			$ucid = Contact::getIdForURL($contact['url'], $uid, true);
+		}
+
+		return ['public' => $pcid, 'user' => $ucid];
+	}
+
+	/**
+	 * @brief Block contact id for user id
+	 *
+	 * @param int     $cid     Either public contact id or user's contact id
+	 * @param int     $uid     User ID
+	 * @param boolean $blocked Is the contact blocked or unblocked?
+	 */
+	public static function setBlocked($cid, $uid, $blocked)
+	{
+		$cdata = self::getPublicAndUserContacID($cid, $uid);
+		if (empty($cdata)) {
+			return;
+		}
+
+		if ($cdata['user'] != 0) {
+			DBA::update('contact', ['blocked' => $blocked], ['id' => $cdata['user'], 'pending' => false]);
+		}
+
+		DBA::update('user-contact', ['blocked' => $blocked], ['cid' => $cdata['public'], 'uid' => $uid], true);
+	}
+
+	/**
+	 * @brief Returns "block" state for contact id and user id
+	 *
+	 * @param int $cid Either public contact id or user's contact id
+	 * @param int $uid User ID
+	 *
+	 * @return boolean is the contact id blocked for the given user?
+	 */
+	public static function getBlocked($cid, $uid)
```

Comment:
Since this function returns a boolean and depends on a user Id, please rename to `isBlockedByUser`.